mne_icalabel/gui/_label_components.py

```diff
@@ -0,0 +1,278 @@
+        self.widget_topo = TopomapFig()
+        self.widget_psd = PowerSpectralDensityFig()
+        self.widget_timeSeries = TimeSeriesFig()
+
+        # add central widget and layout
+        self.central_widget = QWidget(self)
+        self.central_widget.setObjectName("central_widget")
+        layout = QGridLayout()
+        layout.addWidget(self.list_components, 0, 0, 2, 1)
+        layout.addWidget(self.buttonGroup_labels, 0, 1, 2, 1)
+        layout.addWidget(self.widget_topo, 0, 2)
+        layout.addWidget(self.widget_psd, 0, 3)
+        layout.addWidget(self.widget_timeSeries, 1, 2, 1, 2)
+        self.central_widget.setLayout(layout)
+        self.setCentralWidget(self.central_widget)
+        self.resize(1500, 600)
+
+        # dictionary to remember selected labels, with the key
+        # as the 'label' and the values as list of ICA components
+        self.saved_labels: Dict[str, List] = dict()
+
+        # connect signal and slots
+        self.connect_signals_to_slots()
+
+    @staticmethod
+    def list_components(ica):
+        """List the components in a QListView."""
+        list_components = QListWidget()
+        list_components.setSelectionMode(QAbstractItemView.SingleSelection)
+        list_components.setMinimumWidth(6 * _CH_MENU_WIDTH)
+        list_components.setMaximumWidth(6 * _CH_MENU_WIDTH)
+        list_components.addItems([f"ICA{str(k).zfill(3)}" for k in range(ica.n_components_)])
+        return list_components
+
+    def connect_signals_to_slots(self):  # noqa: D102
+        # connect click to function
+        self.list_components.clicked.connect(self.list_component_clicked)
+
+        # TODO: connect selection (i.e. with up/down arrow) to function
+        # self.list_components.currentIndexChanged.connect(self.list_component_clicked)
+        self.buttonGroup_labels.buttonGroup.buttons()[-1].clicked.connect(self.reset)
+
+    @Slot()
+    def list_component_clicked(self):
+        """Jump to the selected component and draw the plots."""
+        self.update_saved_labels()
+        self._reset_all_buttons()
+
+        # reset figures
+        self.widget_topo.reset()
+        self.widget_psd.reset()
+
+        # update selected IC
+        self._current_ic = self.list_components.currentRow()
+
+        # create dummy axes
+        dummy_fig, dummy_axes = plt.subplots(3)
+        axes = [
+            self.widget_topo.axes,
+            dummy_axes[0],
+            dummy_axes[1],
+            self.widget_psd.axes,
+            dummy_axes[2],
+        ]
+        self._ica.plot_properties(self._inst, axes=axes, picks=self._current_ic, show=False)
+        del dummy_fig
+
+        # update figures
+        self.widget_topo.redraw()
+        self.widget_psd.redraw()
+        # retrieve layout and swaap timeSeries widget
+        self.central_widget.layout().replaceWidget(
+            self.widget_timeSeries,
+            self._ica.plot_sources(self._inst, picks=[self._current_ic]),
+        )
```

Comment:
Things to improve...

Reply:
Yes, the structure has to be reworked, for now, it's mostly to test it out.

Reply:
Just adding a comment to link to the code in the discourse :p 

https://mne.discourse.group/t/embedding-time-series-plot-of-mne-qt-browser-into-our-own-gui/5090